Source/AwsCommonRuntimeKit/io/StreamCore.swift

```diff
@@ -0,0 +1,95 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import AwsCIo
+import AwsCCommon
+
+/// aws_input_stream has acquire and release functions which manage the lifetime of this object.
+class IStreamCore {
+    var rawValue: aws_input_stream
+    let iStreamable: IStreamable
+    var isEndOfStream: Bool = false
+    private let allocator: Allocator
+    private var vtable = aws_input_stream_vtable(seek: doSeek,
+            read: doRead,
+            get_status: doGetStatus,
+            get_length: doGetLength,
+            acquire: { _ = Unmanaged<IStreamCore>.fromOpaque($0!.pointee.impl).retain() },
+            release: { Unmanaged<IStreamCore>.fromOpaque($0!.pointee.impl).release() })
+    private let vtablePointer: UnsafeMutablePointer<aws_input_stream_vtable>
+
+    init(iStreamable: IStreamable, allocator: Allocator) {
+        self.allocator = allocator
+        self.iStreamable = iStreamable
+        rawValue = aws_input_stream()
+        // Use a manually managed vtable pointer to avoid undefined behavior
+        self.vtablePointer = allocator.allocate(capacity: 1)
+        vtablePointer.initialize(to: vtable)
+        rawValue.vtable = UnsafePointer(vtablePointer)
+
+        rawValue.impl = Unmanaged<IStreamCore>.passUnretained(self).toOpaque()
+    }
+
+    deinit {
+        allocator.release(vtablePointer)
+    }
+}
+
+private func doSeek(_ stream: UnsafeMutablePointer<aws_input_stream>!,
+                    _ offset: Int64,
+                    _ seekBasis: aws_stream_seek_basis) -> Int32 {
+    let iStreamCore = Unmanaged<IStreamCore>.fromOpaque(stream.pointee.impl).takeUnretainedValue()
+    let iStreamable = iStreamCore.iStreamable
+    do {
+        let streamSeekType = StreamSeekType(rawValue: seekBasis.rawValue)!
+        try iStreamable.seek(offset: offset, streamSeekType: streamSeekType)
+        iStreamCore.isEndOfStream = false
+        return AWS_OP_SUCCESS
+    } catch CommonRunTimeError.crtError(let crtError) {
+        return aws_raise_error(crtError.code)
+    } catch {
+        return aws_raise_error(Int32(AWS_IO_STREAM_SEEK_FAILED.rawValue))
+    }
+}
+
+private func doRead(_ stream: UnsafeMutablePointer<aws_input_stream>!,
+                    _ buffer: UnsafeMutablePointer<aws_byte_buf>!) -> Int32 {
```

Comment:
Nit: More Swifty format

```suggestion
private func doRead(
    _ stream: UnsafeMutablePointer<aws_input_stream>!,
    _ buffer: UnsafeMutablePointer<aws_byte_buf>!
) -> Int32 {
```

Reply:
Thanks, I will update it in a separate formatting PR.